aws/resource_aws_sagemaker_endpoint_configuration.go

```diff
@@ -0,0 +1,278 @@
+		return err
+	}
+	if err := d.Set("kms_key_id", endpointConfig.KmsKeyId); err != nil {
+		return err
+	}
+
+	tagsOutput, err := conn.ListTags(&sagemaker.ListTagsInput{
+		ResourceArn: endpointConfig.EndpointConfigArn,
+	})
+	if err != nil {
+		return fmt.Errorf("error listing tags of Sagemaker endpoint configuration %s: %s", d.Id(), err)
+	}
+	if err := d.Set("tags", tagsToMapSagemaker(tagsOutput.Tags)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func resourceAwsSagemakerEndpointConfigurationUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	d.Partial(true)
+
+	if err := setSagemakerTags(conn, d); err != nil {
+		return err
+	} else {
+		d.SetPartial("tags")
+	}
+
+	d.Partial(false)
+
+	return resourceAwsSagemakerEndpointConfigurationRead(d, meta)
+}
+
+func resourceAwsSagemakerEndpointConfigurationDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	deleteOpts := &sagemaker.DeleteEndpointConfigInput{
+		EndpointConfigName: aws.String(d.Id()),
+	}
+	log.Printf("[INFO] Deleting Sagemaker endpoint configuration: %s", d.Id())
+
+	return resource.Retry(5*time.Minute, func() *resource.RetryError {
+		_, err := conn.DeleteEndpointConfig(deleteOpts)
+		if err == nil {
+			return nil
+		}
+
+		sagemakerErr, ok := err.(awserr.Error)
+		if !ok {
+			return resource.NonRetryableError(err)
+		}
+
+		if sagemakerErr.Code() == "ResourceNotFound" {
+			return resource.RetryableError(err)
+		}
+
+		return resource.NonRetryableError(fmt.Errorf("Error deleting Sagemaker endpoint configuration: %s", err))
+	})
+}
+
+func expandProductionVariants(configured []interface{}) ([]*sagemaker.ProductionVariant, error) {
+	containers := make([]*sagemaker.ProductionVariant, 0, len(configured))
+
+	for _, lRaw := range configured {
+		data := lRaw.(map[string]interface{})
+
+		l := &sagemaker.ProductionVariant{
+			InstanceType:         aws.String(data["instance_type"].(string)),
+			ModelName:            aws.String(data["model_name"].(string)),
+			InitialInstanceCount: aws.Int64(int64(data["initial_instance_count"].(int))),
+		}
+
+		if v, ok := data["variant_name"]; ok {
+			l.VariantName = aws.String(v.(string))
+		} else {
+			l.VariantName = aws.String(resource.UniqueId())
+		}
+
+		if v, ok := data["initial_variant_weight"]; ok {
+			l.InitialVariantWeight = aws.Float64(v.(float64))
+		} else {
+			l.InitialVariantWeight = aws.Float64(1)
+		}
+
+		if v, ok := data["accelerator_type"]; ok && v.(string) != "" {
+			l.AcceleratorType = aws.String(data["accelerator_type"].(string))
+		}
+
+		containers = append(containers, l)
+	}
+
+	return containers, nil
+}
+
+func flattenProductionVariants(list []*sagemaker.ProductionVariant) []map[string]interface{} {
+	result := make([]map[string]interface{}, 0, len(list))
+
+	for _, i := range list {
+		l := map[string]interface{}{
+			"instance_type":          *i.InstanceType,
```

Comment:
To prevent potential panics, we should prefer to use the [AWS Go SDK provided dereferencing functions](https://docs.aws.amazon.com/sdk-for-go/api/aws/) here and below. We can likely also remove `nil` checks below as well with them. e.g.

```suggestion
			"instance_type":          aws.StringValue(i.InstanceType),
```